### ISSUE TITLE
Although I'm using string_val_style='plain', a string value is still quoted

### ISSUE THREAD

**Issue by mnieber** [mk-fg/pretty-yaml#39]
```
x = {'a':['hello world', 'hi']}
print(pyaml.dump(x, string_val_style='plain'))
```

outputs

```
a:
  - 'hello world'
  - hi
```

whereas I was expecting

```
a:
  - hello world
  - hi
```

**Comment by mk-fg** [comment]
Iirc problem is that style returned by choose_scalar_style is also used in mapping keys, and those don't work with spaces in them, and presumably I didn't find any quick and easy hack to make that method work differently for actual values (without overriding or monkey-patching some large chunks of pyyaml).

Will check it out though, maybe just haven't had time to look into it for more than couple minutes at the time.
Thanks for bringing it up.

**Comment by mk-fg** [comment]
Indeed it wasn't that difficult to detect when such check is made in the context of the mapping key and avoid changing style for all values with spaces that way.
Should be fixed in 21.10.1.
